### DISCUSSION TITLE
Cached Tools Documentation

### DISCUSSION THREAD

**Discussion by bxtp4p** [community/community#160668] [Actions]
Can some documentation on how Cached Tools is used be provided? For example, it seems like it's used by various `setup-*` actions but is it used for other things? And what happens if someone tries to install something directly (e..g, via `apt get install`) for something that is in Cached Tools? Will that use the cached version then?

**Comment by vpolikarpov-akvelon** [comment]
Hi @bxtp4p. Tools cache is an instrument that allows downloading and caching tools in runners. This is useful if you want to switch back and forth between versions of a tool, or save a tool between runs (in case of self-hosted runners). The core element of this instrument is [@actions/tool-cache](https://www.npmjs.com/package/@actions/tool-cache) library. It's used in `setup-*` actions indeed (e.g. [setup-go](https://github.com/actions/setup-go/blob/main/src/installer.ts#L86)).

Speaking of GitHub-hosted runners, some tools are preinstalled in images to the place where tool-cache library expects them to be for `setup-*` actions to complete fast for most popular versions of supported tools.

But when you install something 'directly', tool cache is not used since vanilla utils don't know anything about this mechanics.

**Comment by bxtp4p** [comment]
Thanks @vpolikarpov-akvelon - this is helpful. Does it make sense to include at least some of this info in the README for the runner images?

**Reply by vpolikarpov-akvelon** [reply]
There are some information about tool cache in official GitHub documentation, for example:
- [How to use tool-cache for your own CLI tool](https://docs.github.com/en/actions/creating-actions/developing-a-third-party-cli-action)
- [How to use cached Python](https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#specifying-a-python-version)
- [How to use tool cache on self-hosted runners](https://docs.github.com/en/admin/github-actions/managing-access-to-actions-from-githubcom/setting-up-the-tool-cache-on-self-hosted-runners-without-internet-access)

**Reply by bxtp4p** [reply]
Thanks for those links. I think what we're looking for is more around general documentation. Not necessarily "how" to use it but "what" it is and how it's generally used by Actions workflows.

Anyways, just some feedback I've received and wanted to share. Thanks for your responses.
